mito-ai/src/utils/models.ts

```diff
@@ -0,0 +1,96 @@
+import type { IInlineCompletionItem } from '@jupyterlab/completer';
+import type OpenAI from 'openai';
+
+export type ConnectionMessage = {
+  type: 'connection';
+  client_id: string;
+};
+
+export type CompletionError = {
+  type: string;
+  title: string;
+  traceback: string;
+};
+
+/**
+ * Mito AI completion request.
+ */
+export interface ICompletionRequest {
+  /**
+   * The type of the message.
+   */
+  type: string;
+  /**
+   * The message ID.
+   */
+  message_id: string;
+  /**
+   * The chat completion messages history.
+   */
+  messages: OpenAI.Chat.ChatCompletionMessageParam[];
+  /**
+   * Whether to stream the completion or not.
+   */
+  stream: boolean;
+}
+
```

Comment:
Or maybe even better is `IWebsocketMessageRequest` because we also are heavily using the word `message`

Reply:
I kept the word `completion` because it is the wording used by the OpenAI API we are calling. As in the future we may call other kind of API, it felt better than a too generic `MessageRequest`. Let's discuss that.